docs/images.md

```diff
@@ -5,31 +5,157 @@ You create your Docker image and push it to a registry before referring to it in
-Kubernetes has native support for the [Google Container Registry](https://cloud.google.com/tools/container-registry/), when running on Google Compute Engine.  If you are running your cluster on Google Compute Engine or Google Container Engine, simply use the full image name (e.g. gcr.io/my_project/image:tag) and the kubelet will automatically authenticate and pull down your private image.
+
+Kubernetes has native support for the [Google Container
+Registry](https://cloud.google.com/tools/container-registry/), when running on Google Compute
+Engine (GCE).  If you are running your cluster on GCE or Google Container Engine (GKE), simply
+use the full image name (e.g. gcr.io/my_project/image:tag) and the kubelet will automatically
+authenticate and pull down your private image.
+
+All pods in a cluster will have read access to images in this registry.
 
 ### Other Private Registries
-Docker stores keys for private registries in a `.dockercfg` file.  Create a config file by running `docker login <registry>.<domain>` and then copying the resulting `.dockercfg` file to the kubelet working dir.
-The kubelet working dir varies by cloud provider.  It is `/` on GCE and `/home/core` on CoreOS.  You can determine the working dir by running this command:
-`sudo ls -ld /proc/$(pidof kubelet)/cwd` on a kNode.
+Docker stores keys for private registries in a `.dockercfg` file.  Create a config file by running
+`docker login <registry>.<domain>` and then copy the resulting `.dockercfg` file to the kubelet
+working dir on each node in your cluster.  The kubelet working dir varies by cloud provider.  It is `/` on GCE and `/home/core`
+on CoreOS.  You can determine the working dir by running this command: `sudo ls -ld /proc/$(pidof
+kubelet)/cwd` on a node.
+
+You must ensure all nodes in the cluster have the same `.dockercfg`.  Otherwise, pods will run on
+some nodes and fail to run on others.  For example, if you use node autoscaling, then each instance
+template needs to include the `.dockercfg` or mount a drive that contains it.
 
-All users of the cluster will have access to any private registry in the `.dockercfg`.
+All pods will have read access to images in any private registry with keys in the `.dockercfg`.
 
-## Preloading Images
+### Pre-pulled Images
 
 Be default, the kubelet will try to pull each image from the specified registry.
 However, if the `imagePullPolicy` property of the container is set to `IfNotPresent` or `Never`,
 then a local image is used (preferentially or exclusively, respectively).
 
+If you want to rely on pre-pulled images as a substitute for registry authentication,
+you must ensure all nodes in the cluster have the same pre-pulled images.
+
 This can be used to preload certain images for speed or as an alternative to authenticating to a private registry.
 
-Pull Policy is per-container, but any user of the cluster will have access to all local images.
+All pods will have read access to any pre-pulled images.
```

Comment:
> On Jun 11, 2015, at 7:41 AM, David Eads <notifications@github.com> wrote:
> 
> In docs/images.md:
> 
> >  This can be used to preload certain images for speed or as an alternative to authenticating to a private registry.
> >  
> > -Pull Policy is per-container, but any user of the cluster will have access to all local images.
> > +All pods will have read access to any pre-pulled images.
> Are you saying that if a container uses PullIfNotPresent, that it potentially has access to all images ever pulled on the machine, including those pulled with imagePullSecret?
> 
> Yes, this is a side-effect of not having the equivalent of the Docker registry not having a SubjectAccessReview (MotherMayI) API. It will work for the same reason that pre-pulling images will work.
> 
> @smarterclayton Suggested a couple workarounds, but they all behaved unexpectedly when access to the Docker registry was down, but not denied. Things like "try to pull the image and make sure you start getting a layer back".
> 
We should probably work with upstream to get a HEAD check for is the current client authorized to pull a given image
> —
> Reply to this email directly or view it on GitHub.
>